frontend/src/queries/nodes/DataTable/DataTableExport.tsx

```diff
@@ -39,12 +43,141 @@ function startDownload(query: DataTableNode, onlySelectedColumns: boolean): void
     })
 }
 
+const columnDisallowList = ['person.$delete', '*']
+const getCsvTableData = (dataTableRows: DataTableRow[], columns: string[], query: DataTableNode): string[][] => {
+    if (isPersonsNode(query.source)) {
+        const filteredColumns = columns.filter((n) => !columnDisallowList.includes(n))
+
+        const csvData = dataTableRows.map((n) => {
+            const record = n.result as Record<string, any> | undefined
+            const recordWithPerson = { ...(record ?? {}), person: record?.name }
+
+            return filteredColumns.map((n) => recordWithPerson[n])
+        })
+
+        return [filteredColumns, ...csvData]
+    }
+
+    if (isEventsQuery(query.source)) {
+        const filteredColumns = columns
+            .filter((n) => !columnDisallowList.includes(n))
+            .map((n) => extractExpressionComment(n))
+
+        const csvData = dataTableRows.map((n) => {
+            return columns
+                .map((col, colIndex) => {
+                    if (columnDisallowList.includes(col)) {
+                        return null
+                    }
+
+                    if (col === 'person') {
+                        return n.result?.[colIndex]?.properties?.email || n.result?.[colIndex]?.uuid
+                    }
+
+                    return n.result?.[colIndex]
+                })
+                .filter(Boolean)
+        })
+
+        return [filteredColumns, ...csvData]
+    }
+
+    if (isHogQLQuery(query.source)) {
+        return [columns, ...dataTableRows.map((n) => (n.result as any[]) ?? [])]
+    }
+
+    return []
+}
+
+const getJsonTableData = (
+    dataTableRows: DataTableRow[],
+    columns: string[],
+    query: DataTableNode
+): Record<string, any>[] => {
+    if (isPersonsNode(query.source)) {
+        const filteredColumns = columns.filter((n) => !columnDisallowList.includes(n))
+
+        return dataTableRows.map((n) => {
+            const record = n.result as Record<string, any> | undefined
+            const recordWithPerson = { ...(record ?? {}), person: record?.name }
+
+            return filteredColumns.reduce((acc, cur) => {
+                acc[cur] = recordWithPerson[cur]
+                return acc
+            }, {} as Record<string, any>)
+        })
+    }
+
+    if (isEventsQuery(query.source)) {
+        return dataTableRows.map((n) => {
+            return columns.reduce((acc, col, colIndex) => {
+                if (columnDisallowList.includes(col)) {
+                    return acc
+                }
+
+                if (col === 'person') {
+                    acc[col] = n.result?.[colIndex]?.properties?.email || n.result?.[colIndex]?.uuid
+                    return acc
+                }
```

Comment:
These person checks could be unified. There's also a [function](https://github.com/PostHog/posthog/blob/8c046d74e75ebecf65e80f95fd1fa796aed23703/frontend/src/scenes/persons/person-utils.ts#L36) `asDisplay` that could be used here.